it/uk/gov/hmrc/soletraderidentificationfrontend/api/controllers/JourneyControllerISpec.scala

```diff
@@ -48,6 +48,21 @@ class JourneyControllerISpec extends ComponentSpecHelper with JourneyStub with S
       await(journeyConfigRepository.findById(testJourneyId)) mustBe Some(testSoleTraderJourneyConfig)
     }
 
+    "support a json optFullNamePageLabel field" in {
+      stubAuth(OK, successfulAuthResponse())
+      stubCreateJourney(CREATED, Json.obj("journeyId" -> testJourneyId))
+
+      post(
+        uri = "/sole-trader-identification/api/journey",
+        json = testSoleTraderJourneyConfigJson + ("optFullNamePageLabel" -> JsString(testFullNamePageLabel))
+      )
+
+      val expectedSoleTraderJourneyConfig = testSoleTraderJourneyConfig
+        .copy(pageConfig = testSoleTraderPageConfig.copy(optFullNamePageLabel = Some(testFullNamePageLabel)))
+
+      await(journeyConfigRepository.findById(testJourneyId)) mustBe Some(expectedSoleTraderJourneyConfig)
+    }
```

Comment:
I think these tests would read better like

"POST /api/journey" should {
    "return the correct journeyStartUrl" when {
        "the optFullNamePageLabel field is provided" in {
          }
        "the optFullNamePageLabel field is not provided" in {
          }
      }
    "redirect to Sign In page" when {
        "the user is UNAUTHORISED" in {
         }
    }
}

Reply:
Hi Milly, I did not want to touch the existing one that is pointing out ... "redirect to Capture Full Name Controller". 
1) Do you want me to rename that test ??? 
2) Or we can rename like in the screenshot!
3) Or I leave the "redirect to Capture Full Name Controller" and I added other 2 tests related to optFullNamePageLabel??? ... the content of the body of one of them (the one without the optFullNamePageLabel) will be the same of the original "redirect to Capture Full Name Controller"

attached you can see the original and a renaming suggested in point 2 

<img width="747" alt="Screenshot 2022-01-05 at 09 28 53" src="https://user-images.githubusercontent.com/4958044/148194251-32921efe-633a-4ab7-a41f-a6b40dd7a2e4.png">
<img width="1011" alt="Screenshot 2022-01-05 at 09 28 42" src="https://user-images.githubusercontent.com/4958044/148194503-a8ac5c21-9336-46bf-be7f-ba1b90176d9b.png">

Reply:
The screenshot rename is fine. Just a lot of duplication in test name.

Renaming the excising test would have been fine though, it's not named the best because it doesn't actually redirect to the full name page. It returns a 201 with the continue url in the json

Reply:
okay I renamed it ... what about:

<img width="719" alt="Screenshot 2022-01-05 at 10 08 46" src="https://user-images.githubusercontent.com/4958044/148199930-8e945f2b-5e9d-43b4-993b-ce399690338f.png">

Reply:
Perfect